rfc/2020/conformance-validation/mayankshah1607/rfc.md

```diff
@@ -0,0 +1,449 @@
+    values: 
+        # linkerd install flags that hold values
+        # useful for properties like external certs, mtls
+testOptions:
+    # duration for which tests must wait for pods to come to running state
+    wait: <int>
+    
+    # lookback time used for tracing
+    tracingLookback: <String>
+    
+    # service used for tracing
+    tracingService: <String>
+
+    # array of test names that must be executed
+    # If empty, run all tests
+    enableTests: <array>
+
+    # whether to inject MySQL pod 
+    injectMySQL: <bool>                                                                     
+
+    # whether to inject Redis pods
+    injectRedis: <bool>
+   
+   # users may be allowed to run these tests on
+   # services other than the default movieChat by 
+   # mentioning the namespace in which they are deployed
+   # However, protocol / topology related tests might fail
+   # if running on non-default namespace
+   # GOOD-TO-HAVE feature
+   workloadNamespace: <String> 
+
+   # RBAC configuration to be used if on GKE
+   # If left empty, use default RBAC settings as
+   # mentioned here
+   gkeRbacConfig: <string>
+
+```
+
+This is great for things like setting installation options and additional test preferences such as which tests to run, wait time, lookback time for tracing, whether to inject MySQL / Redis pods, etc. An example configuration file :
+```yaml
+installOptions:
+    bools:
+        - control-plane-tracing
+        - ha
+        - disable-h2-upgrade
+    values:
+        identity-external-issuer: ./certs/ca.crt
+        identity-issuer-certificate-file : ./certs/issuer.crt
+        identity-issuer-key-file: issuer.key
+testOptions:
+    wait: 10000
+    tracingLookback: 1h
+    tracingService: nginx
+    enableTests:
+        - proxy-injection
+        - tap
+        - distributed-tracing
+        - canary
+        - ingress
+    injectMySQL: true
+    injectRedis: true
+    workloadNamespace: moviechat
+    gkeRbacConfig: ./path/to/config
+
+```
+
+### Testing Methodologies for various features
+The conformance test suite is intended to be run against an installation of Linkerd. Passing this test suite would give the user the assurance that a given configuration of Linkerd works (as expected) with a given version of Kubernetes and its cluster configuration. Users may easily configure install options and test preference using the test configuration file provided.
+
+This section includes write ups regarding testing methodologies for some primary features:
+
+**1. Auomatic Proxy Injection**
+
+Proxy injection process works by adding a `linkerd.io/inject: enabled` annotation to pod template / namespace. This triggers an admission webhook that injects the `linkerd-proxy` container to the targeted deployments. This process shall be tested in 7 steps:
+- Check if `inject` command (with and without `--manual` flag) outputs the desired configuration with the appended annotation.
```

Comment:
In case of emojivoto, comparing against a predefined YAML seems fine. But if we're moving outside of emojivoto, we have to use the `client-go` library. Added a note on this, please let me know if that resolves it. :smile: